superset/utils/core.py

```diff
@@ -1578,12 +1578,15 @@ def split(
     yield string[i:]
 
 
-def get_iterable(x: Any) -> list[Any]:
+T = TypeVar("T")
+
+
+def get_as_list(x: T | list[T]) -> list[T]:
```

Comment:
Maybe just `as_list`?

Reply:
I think either `as_iterable` or `as_list` is likely best.

Reply:
@john-bodley `as_iterable` is super confusing, because there are many things that are already iterable but will still be modified by this function, eg:

```python
>>> from typing import Any
>>> def is_iterable(foo: Any) -> bool:
...     try:
...         iter(foo)
...     except Exception:
...         return False
...     return True
...
>>>
>>> a = (1,2,3)
>>> is_iterable(a)
True
>>> as_iterable(a)
[(1, 2, 3)]
```

Reply:
And to be clear, I think this function could be called `as_iterable`, but then it should be improved to not check if the passed object is a lit, but instead use a modified version of the `is_iterable` function above, something like this:

```python
def is_iterable(foo: Any) -> bool:
    if isinstance(foo, str):
        return False
    try:
        iter(foo)
    except Exception:
        return False
    return True
```

But that might introduce bugs. In this PR I just want to fix the erroneous name.